INLConfig/INLConfigGenerator/main.swift

```diff
@@ -0,0 +1,247 @@
+        + "@implementation INLConfig (\(configName))\n\n"
+        + "+(INLConfig *)\(configName.decapitalizedString) {\n\tinl_loadConfig(@\"\(configName)\")\n}\n\n"
+        + config.varGenReduce { key, value, type, name in
+            "-(\(type) *)\(name) {\n"
+                + "\treturn [self \(getterForValue(value)):@\"\(key)\"];\n"
+                + "}\n\n"
+        }
+        + "// Convenience\n"
+        + config.varGenReduce { key, value, type, name in
+            "+(\(type) *)\(name) {\n"
+                + "\treturn [[self \(configName.decapitalizedString)] \(name)];\n"
+                + "}\n\n"
+        }
+        + "@end"
+}
+
+func genSwiftForConfig(_ config: NSDictionary, configName: String, isJson: Bool) -> String {
+    var result = "\n//! Automatically generated file. Do not modify!\n\n"
+        + "extension INLConfig {\n\n"
+        + "\tstatic var \(configName.decapitalizedString): INLConfig {\n"
+        + "\t\tstruct Static {\n"
+    if isJson {
+        result += "\t\t\tstatic let config: INLConfig = INLConfig()\n"
+            + "\t\t\tconfig.loadConfiguration(\"\(configName)\")\n"
+    } else {
+        result += "\t\t\tstatic let config: INLConfig = INLConfig(plist: \"\(configName)\")\n"
+    }
+    result += "\t\t}\n\t\treturn Static.config\n"
+        + "\t}\n"
+        + config.varGenReduce { key, value, type, name in
+            "\n\tvar \(name): \(type) {\n"
+                + "\t\treturn \(getterForValue(value))(\"\(key)\")!\n"
+                + "\t}\n"
+        }
+        + "\n\t// Convenience"
+        + config.varGenReduce { key, value, type, name in
+            "\n\tstatic var \(name): \(type) {\n"
+                + "\t\treturn \(configName.decapitalizedString).\(name)\n"
+                + "\t}\n"
+        }
+        + "}"
+    return result
+}
+
+func pathForFile(_ file: String, root: String) -> String? {
+    let fileManager = FileManager.default
+    
+    if fileManager.fileExists(atPath: "\(root)/\(file)") {
+        return root
+    }
+    if let enumerator = fileManager.enumerator(atPath: root) {
+        for subDir in enumerator {
+            if (subDir as AnyObject).hasSuffix(file) {
+                return (root as NSString).appendingPathComponent((subDir as AnyObject).deletingLastPathComponent)
+            }
+        }
+    }
+    return nil
+}
+
+extension String {
+    var decapitalizedString: String {
+        return String(characters.prefix(1)).lowercased() + String(characters.dropFirst())
+    }
+    func trim() -> String {
+        return trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+    }
+}
+
+func typeForValue(_ value: AnyObject) -> String {
+    if value is String { return "NSString" }
+    if value is NSNumber { return "NSNumber" }
+    if value is NSArray { return "NSArray" }
+    if value is NSDictionary { return "NSDictionary" }
+    if value is Data { return "NSData" }
+    assert(false, "Invalid type \(value)")
+    return "Any"
```

Comment:
I assume yes, but I see it reasonable for Obj-C code generation ? What do you think ?